hypershift-operator/controllers/hostedcluster/internal/platform/kubevirt/kubevirt.go

```diff
@@ -52,6 +54,10 @@ func (p Kubevirt) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, cre
 		return nil, err
 	}
 
+	if _, err := createOrUpdate(ctx, c, composeEgressFirewall(controlPlaneNamespace), func() error { return nil }); err != nil {
+		return nil, err
+	}
+
 	return kubevirtCluster, nil
```

Comment:
unfortunately this won't work for the external worker use case. we'll need to generate a client for the infra cluster that the workers run on.

for example, search for this function `generateKubevirtInfraClusterClient`. If external infra is in use, a different client needs to be used. 

We might be better off with moving this logic to the hostedcluster_controller.go and calling a function called reconcileEgressRules() somewhere in that reconcile loop, which then detects if the KubevirtPlatform is in use and creates the egress rules.

Reply:
Added a `kubevirtInfraCluster` structure to the Reconciler it's initialized at the beginning and used during the resto of reconcile loop.